pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -304,6 +304,7 @@ func (r *ReconcilePerconaServerMongoDBRestore) getBackup(ctx context.Context, cr
 				Namespace: cr.Namespace,
 			},
 			Spec: psmdbv1.PerconaServerMongoDBBackupSpec{
+				Type:        cr.Spec.BackupSource.Type,
```

Comment:
From what I understand, adding this is mostly for consistency, right? Couldn't understand how we use it with this PR's changes.

Reply:
no, this is a bug fix. after incremental backup changes, physical restores using backup source was broken and this fixes it